testutils/resourcebuilder/machine/v1beta1/aws_provider_spec.go

```diff
@@ -127,26 +169,124 @@ func (m AWSProviderSpecBuilder) BuildRawExtension() *runtime.RawExtension {
 	}
 }
 
+// WithAMI sets the AMI for the AWS machine config builder.
+func (m AWSProviderSpecBuilder) WithAMI(ami machinev1beta1.AWSResourceReference) AWSProviderSpecBuilder {
+	m.ami = &ami
+	return m
+}
+
 // WithAvailabilityZone sets the availabilityZone for the AWS machine config builder.
 func (m AWSProviderSpecBuilder) WithAvailabilityZone(az string) AWSProviderSpecBuilder {
-	m.availabilityZone = az
+	m.availabilityZone = &az
+	return m
+}
+
+// WithBlockDevices sets the BlockDevices for the AWS machine config builder.
+// Use WithBlockDevices(nil), to set them to the zero value.
+func (m AWSProviderSpecBuilder) WithBlockDevices(blockDevices []machinev1beta1.BlockDeviceMappingSpec) AWSProviderSpecBuilder {
+	m.blockDevices = &blockDevices
+	return m
+}
+
+// WithCredentialsSecret sets the DeviceIndex for the AWS machine config builder.
+func (m AWSProviderSpecBuilder) WithDeviceIndex(deviceIndex int64) AWSProviderSpecBuilder {
```

Comment:
Can we make sure that all of the functions are alphabetically ordered, and have the correct godoc, this one doesn't for example